static/src/javascripts/projects/commercial/modules/header-bidding/prebid/prebid.js

```diff
@@ -100,13 +100,26 @@ class PrebidAdUnit {
     code: ?string;
     bids: ?(PrebidBid[]);
     mediaTypes: ?PrebidMediaTypes;
+    build: Promise<PrebidAdUnit>;
+
+    constructor(
+        advert: Advert,
+        slot: HeaderBiddingSlot,
+        asyncBids: ?(PrebidBid[])
+    ) {
+        if (typeof asyncBids === 'undefined')
+            throw new Error('Needs to be called async');
 
-    constructor(advert: Advert, slot: HeaderBiddingSlot) {
         this.code = advert.id;
-        this.bids = bids(advert.id, slot.sizes);
+        this.bids = asyncBids;
         this.mediaTypes = { banner: { sizes: slot.sizes } };
     }
 
+    static async build(advert: Advert, slot: HeaderBiddingSlot) {
+        const asyncBids = await bids(advert.id, slot.sizes);
+        return new PrebidAdUnit(advert, slot, asyncBids);
+    }
```

Comment:
This `async` constructor is a little different than the original one. To use the object, we now need to wait for `PrebidAdUnit.build()` to resolve.